library/rsa.c

```diff
@@ -502,10 +514,8 @@ int mbedtls_rsa_set_padding( mbedtls_rsa_context *ctx, int padding,
     if( ( padding == MBEDTLS_RSA_PKCS_V21 ) &&
         ( hash_id != MBEDTLS_MD_NONE ) )
     {
-        const mbedtls_md_info_t *md_info;
-
-        md_info = mbedtls_md_info_from_type( hash_id );
-        if( md_info == NULL )
+        /* Just make sure this hash is supported in this build. */
+        if( mbedtls_hash_info_get_size( hash_id ) == 0 )
```

Comment:
Minor:
```suggestion
        if( mbedtls_hash_info_psa_from_md( hash_id ) == PSA_ALG_NONE )
```
seems more appropriate

Reply:
They're equivalent in practice, but I agree checking for `NONE` looks a bit better.

Reply:
Fixed as suggested.